src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/MemoryCache.cs

```diff
@@ -43,11 +43,16 @@ public bool TryGetValue(TKey key, out TValue result)
 
         public void Set(TKey key, TValue value)
         {
-            if (_dict.Count >= SizeLimit)
+            if (_dict.Count >= _sizeLimit)
             {
                 Compact();
             }
 
+            if (_dict.ContainsKey(key))
```

Comment:
Was forced to add this in due to exceptions being thrown when you add the same key twice:
https://dev.azure.com/dnceng/public/_build/results?buildId=757405&view=ms.vss-test-web.build-test-results-tab&runId=23507082&resultId=100079&paneView=debug

> System.ArgumentException : The key already existed in the dictionary.